### PR TITLE
[cleanup] Minor changes to test_env_builder

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Minor cleanup, followup from PR https://github.com/near/nearcore/pull/9664

### CODE REVIEW

File: chain/client/src/test_utils/test_env_builder.rs [hunk 1]
```diff
@@ -254,56 +255,61 @@ impl TestEnvBuilder {
     fn ensure_epoch_managers(self) -> Self {
         let mut ret = self.ensure_stores();
         if ret.epoch_managers.is_some() {
-            ret
```
Comment (shreyan-gupta) [inline chain/client/src/test_utils/test_env_builder.rs:257]: view this with whitespace hidden

File: nearcore/src/test_utils.rs [hunk 2]
```diff
@@ -22,35 +25,31 @@ impl TestEnvNightshadeSetupExt for TestEnvBuilder {
...
-                    &genesis.config,
-                    epoch_manager,
-                    runtime_config,
-                    state_snapshot_enabled,
-                ) as Arc<dyn RuntimeAdapter>
-            })
-            .collect();
-        builder.runtimes(runtimes)
+        let state_snapshot_enabled = self.state_snapshot_enabled();
+        let nightshade_runtime_creator = |home_dir: PathBuf,
```
Comment (shreyan-gupta) [inline nearcore/src/test_utils.rs:34]: The reason why we have this function here and not as part of the original test_util is that NightshadeRuntime is only a part of nearcore module and we don't really have access to it in client module.

Keeping that in mind, it makes more sense to just pass a "nightshade runtime creator/constructor" function to the TestEnvBuilder

File: chain/client/src/test_utils/test_env_builder.rs [hunk 3]
```diff
@@ -254,56 +255,61 @@ impl TestEnvBuilder {
...
+        runtime_configs: Vec<RuntimeConfigStore>,
+        nightshade_runtime_creator: impl Fn(
+            PathBuf,
+            Store,
+            Arc<EpochManagerHandle>,
+            RuntimeConfigStore,
+        ) -> Arc<dyn RuntimeAdapter>,
+    ) -> Self {
+        let builder = self.ensure_home_dirs().ensure_epoch_managers().ensure_stores();
+        let runtimes = multizip((
```
Comment (shreyan-gupta) [inline chain/client/src/test_utils/test_env_builder.rs:296]: Now we don't need to pass back 50 things to `nightshade_runtimes_with_runtime_config_store`

**Comment by akhi3030** [comment]
removing myself as I am not actively reviewing this PR.

**Review by wacban — DISMISSED** [review]

**Comment by wacban** [comment]
Looks much nicer, thanks!

**Review by wacban — APPROVED** [review]

**Pull request merged**